packages/reactive-element/src/reactive-element.ts

```diff
@@ -961,14 +1020,23 @@ export abstract class ReactiveElement
    * the native platform default).
    */
   private __saveInstanceProperties() {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const instanceProperties: Array<[keyof this, any]> = [];
     const elementProperties = (this.constructor as typeof ReactiveElement)
       .elementProperties;
-    for (const p of elementProperties.keys()) {
+    for (const p of elementProperties.keys() as IterableIterator<keyof this>) {
       if (this.hasOwnProperty(p)) {
-        this.__instanceProperties!.set(p, this[p as keyof this]);
-        delete this[p as keyof this];
+        instanceProperties.push([p, this[p]]);
+        delete this[p];
       }
     }
+    if (instanceProperties.length) {
```

Comment:
Is this change necessary?

I'm concerned about the potential perf implications (but hopefully this path shouldn't be hit a lot) and additional complexity (e.g. the comment in `_initialize` about ordering wrt `requestUpdate`).

Reply:
This isn't completely necessary, but it eliminates the __instanceProperties field and it's gated by there being instance properties to save. Once standard decorators are the only way to define fields we can do this w/o a microtask in an addInitializer() callback.

Reply:
Per chat, not strictly necessary but preferred since it saves memory and centralizes the code, and this is only called in hopefully a corner case.